openreview/tools.py

```diff
@@ -134,18 +137,29 @@ def get_profiles(client, ids_or_emails, with_publications=False):
 
     profiles = []
     profile_by_email = {}
+    profiles_as_dict = {}
 
     batch_size = 100
     for i in range(0, len(ids), batch_size):
         batch_ids = ids[i:i+batch_size]
         batch_profiles = client.search_profiles(ids=batch_ids)
         profiles.extend(batch_profiles)
 
+    if as_dict:
+        for i, profile in enumerate(profiles):
```

Comment:
is this assuming the profiles are returned in the same order? also what happens with the ids that don't return any profile (it should happen). 

I would build the dictionary using the `ids` arrays instead, same as you are doing with the emails. You would need to find the profile that contains that `id` into the `profile.content['names']` array. Maybe using a dictionary where the key are all the alternate name can help.